src/dmd/console.d

```diff
@@ -57,6 +57,21 @@ struct Console
 
         @property FILE* fp() { return _fp; }
 
+        /**
+        Tries to detect whether DMD has been invoked from a terminal.
+        Returns: `true` if a terminal has been detect, `false` otherwise
+        */
+        static bool detectedTerminal()
+        {
+            auto nStdHandle = STD_OUTPUT_HANDLE;
+            auto h = GetStdHandle(nStdHandle);
+            CONSOLE_SCREEN_BUFFER_INFO sbi;
+            if (GetConsoleScreenBufferInfo(h, &sbi) == 0) // get initial state of console
+                return false;
```

Comment:
This is probably just my personal style, but I like to include a comment for raw boolean return statements,
```D
    return false; // no terminal detected

return true; // terminal detected
```